plugins/gstperf.c

```diff
@@ -401,16 +422,91 @@ gst_perf_cpu_get_load (GstPerf * perf, guint32 * cpu_load)
   } else {
     *cpu_load = 0;
   }
+
   /*Remember the total and idle CPU for the next check */
   perf->prev_cpu_total = total;
   perf->prev_cpu_idle = idle;
+
+  return TRUE;
+
+cpu_failed:
+  GST_ERROR_OBJECT (perf, "Failed to get the CPU load");
+  return FALSE;
+}
+
+#elif IS_MACOSX
+static gboolean
+gst_perf_cpu_get_load_macosx (GstPerf * perf, guint32 * cpu_load)
+{
+  guint32 idle = 0;
+  guint32 total = 0;
+  guint32 diff_total, diff_idle;
+  host_cpu_load_info_data_t cpuinfo;
+  mach_msg_type_number_t count = HOST_CPU_LOAD_INFO_COUNT;
+
+  g_return_val_if_fail (perf, FALSE);
+  g_return_val_if_fail (cpu_load, FALSE);
+
+  /* Default value in case of failure */
+  *cpu_load = -1;
```

Comment:
`cpu_load` is unsigned int, so -1 is equal 2^32

Reply:
What value do you suggest for invalid measurement (for the print)? I thought about zero, but zero is a common CPU usage measurement.

Reply:
Yes, typically `-1` can be used to represent an invalid value. However, the rest of the logic should be handled it and does not operate using that invalid. This was rather an advice.